editoast/src/models/projects.rs

```diff
@@ -231,83 +231,52 @@ impl List<Ordering> for ProjectWithStudies {
-
-    #[actix_test]
-    async fn create_delete_project() {
-        let project = build_test_project();
-        let manager = ConnectionManager::<PgConnection>::new(PostgresConfig::default().url());
-        let pool = Data::new(Pool::builder().max_size(1).build(manager).unwrap());
-
-        // Create a project
-        let project = project.create(pool.clone()).await.unwrap();
-        let project_id = project.id.unwrap();
+    #[rstest]
+    async fn create_delete_project(#[future] project: TestFixture<Project>, db_pool: Data<DbPool>) {
+        let project = project.await;
         // Delete the project
-        assert!(Project::delete(pool.clone(), project_id).await.unwrap());
+        assert!(Project::delete(db_pool.clone(), project.id())
+            .await
+            .unwrap());
         // Second delete should be false
-        assert!(!Project::delete(pool.clone(), project_id).await.unwrap());
+        assert!(!Project::delete(db_pool, project.id()).await.unwrap());
     }
 
-    #[actix_test]
-    async fn get_project() {
-        let project = build_test_project();
-        let manager = ConnectionManager::<PgConnection>::new(PostgresConfig::default().url());
-        let pool = Data::new(Pool::builder().max_size(1).build(manager).unwrap());
-
-        // Create a project
-        let project = project.create(pool.clone()).await.unwrap();
-        let project_id = project.id.unwrap();
+    #[rstest]
+    async fn get_project(#[future] project: TestFixture<Project>, db_pool: Data<DbPool>) {
+        let project = project.await;
 
         // Get a project
-        assert!(Project::retrieve(pool.clone(), project_id)
+        assert!(Project::retrieve(db_pool.clone(), project.id())
             .await
             .unwrap()
             .is_some());
         assert!(
-            ProjectWithStudies::list(pool.clone(), 1, 25, Ordering::LastModifiedAsc)
+            ProjectWithStudies::list(db_pool.clone(), 1, 25, Ordering::LastModifiedAsc)
                 .await
                 .is_ok()
         );
-
-        // Delete the project
-        assert!(Project::delete(pool.clone(), project_id).await.unwrap());
     }
 
-    #[actix_test]
-    async fn sort_project() {
-        let mut project = build_test_project();
-        let manager = ConnectionManager::<PgConnection>::new(PostgresConfig::default().url());
-        let pool = Data::new(Pool::builder().max_size(1).build(manager).unwrap());
+    #[rstest]
+    async fn sort_project(#[future] project: TestFixture<Project>, db_pool: Data<DbPool>) {
+        let project = project.await;
 
-        // Create first project
-        let project_1 = project.clone().create(pool.clone()).await.unwrap();
-        let project_id_1 = project_1.id.unwrap();
         // Create second project
-        project.name = Some("project_test".into());
-        let project_2 = project.create(pool.clone()).await.unwrap();
-        let project_id_2 = project_2.id.unwrap();
+        let project_2 = Project {
```

Comment:
This could also be in the fixtures file, that’s always useful

Reply:
I don't feel like creating a fixture in this case would be useful, we would be factorizing code which is only used once

Reply:
We could imagine a fixture like that:

```rust
#[fixture]
pub async fn projects(#[default(1)] number: i32, db_pool: Data<DbPool>) -> Vec<TestFixture<Project>> {
    // ...
}

#[rstest]
async fn sort_project(#[future] #[with(2)] projects: Vec<TestFixture<Project>>, db_pool: Data<DbPool>) {
   // ...
}
```

I don't think this is required here.